syntax = "proto3";

package geospatial;

import "files.proto";
// =============================================================================
// SOLICITUDES/RESPUESTAS DEL SERVICIO PRINCIPAL
// =============================================================================

// Ejemplo simple de Hello World
message HelloWorldRequest {
  string message = 1;
}

message HelloWorldResponse {
  string message = 1;
}

// Ejemplo de parámetro de echo
message EchoParameterRequest {
  double value = 1;
  string operation = 2;
}

message EchoParameterResponse {
  double original_value = 1;
  double processed_value = 2;
  string operation = 3;
}

// Verificación de salud
message HealthCheckRequest {}

message HealthCheckResponse {
  bool healthy = 1;
  string version = 2;
  map<string, string> status = 3;
}

// =============================================================================
// Formato para entregar datos para visualización
// En este caso, es para datos en dos dimensiones: x,y. z es el valor de la variable.
// =============================================================================

// Lo que envia el frontend para obtener datos
// Se recibe el dato que se quiere como z (data_types)
// el número de puntos máximos a obtener (max_points)

message GetBatchDataRequest {
  repeated string data_types = 2;
  int32 max_points = 3;
}
// Establesco los limites para el gráfico
message Bounds {
  double max_value = 1;
  double min_value = 2;
}

message GetColumnarDataResponse {
  repeated double data = 1;  // Array con posiciones: [x1,y1,z1, x2,y2,z2, x3,y3,z3, ...]
  int32 total_count = 2;     // Numéro de puntos
  // Uso map para poder después establecer el limite del valor, por ejemplo bounds["x"]
  // Podría usar repeated pero tendria que entrar con indices (bounds[0])
  map<string, Bounds> bounds = 3; 
  // Puedo agregar más datos...

}
